### PR TITLE
fixes: Webcam.close() does not return #651

### PR DESCRIPTION
**Pull request by jonnio**
SynchronousQueue can throw InterruptedException if the queue transfer result was `null`. Users should call `Thread.interrupted()` when deciding if the thread was actually interrupted.

### CODE REVIEW

**Comment by sarxos** [comment]
Hi @jonnio,

Thank you for your contribution!

I have a few questions in regards to your pull request. I noticed that you changed atomic execution order and I do not understand why:

```java
Webcam.java
-		if (open.compareAndSet(true, false)) {
+		if (open.get()) {
```

And:

```
Webcam.java
-		return true;
+		return open.compareAndSet(true, false);
```

When I wrote original code my intention was to ensure that code block in `if (open.compareAndSet(true, false))` is executed once and only once. With your change this is not guaranteed. If this compare-and-set is problematic then maybe we should `synchronize` the method instead relying on atomics?

File: pom.xml [hunk 1]
```diff
@@ -160,8 +160,8 @@
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
         <configuration>
-          <source>1.7</source>
-          <target>1.7</target>
+          <source>11</source>
```
Comment (sarxos) [inline pom.xml]: Hi @jonnio. I'm a little bit concerned in regards to changing this from 1.7 to 11. I presume that this change will cause newly compiled release to be compatible with Java 11+?

Comment (jlolling) [reply pom.xml:163]: Can we stick with Java 8 as long as we do not have any dependencies to higher java versions?

**Comment by alexmao86** [comment]
hi @jonnio , how about this merge activity? can we move on?
thanks

**Comment by jonnio** [comment]
HI Alex,

I wouldn’t merge it. I’ve found a couple of other issues. It is a java 11+ problem that we should solve.

 

Thanks,

Jon Osborn

Bell Tracy, Ltd.

http://www.belltracy.com

 

From: Alex Mao ***@***.***>
Reply-To: sarxos/webcam-capture ***@***.***>
Date: Monday, May 31, 2021 at 7:11 AM
To: sarxos/webcam-capture ***@***.***>
Cc: Jon Osborn ***@***.***>, Mention ***@***.***>
Subject: Re: [sarxos/webcam-capture] fixes: Webcam.close() does not return #651 (#697)

 

hi @jonnio , how about this merge activity? can we move on?
thanks

**Comment by alexmao86** [comment]
confirmed via private email. close this loop

**Pull request closed without merging**